### PR TITLE
Turn testing Python 2.6 in Travis on

### PR DESCRIPTION
**Pull request by jeremycline**
Twisted has dropped Python 2.6 support, but fedmsg still needs to be
Python 2.6 compatible. Since moksha.hub, the dependency that pulls in
Twisted, doesn't require a specific version of Twisted, we can just
install it before we install the requirements if the environment is
Python 2.6.

Signed-off-by: Jeremy Cline <jeremy@jcline.org>

### CODE REVIEW

**Comment by jeremycline** [comment]
This pulls in all the commits for #379 - we can either merge this and close #379, or merge #379 and rebase this. I do not mind either way.

File: .travis.yml [hunk 1]
```diff
@@ -4,11 +4,14 @@ python:
  - "2.7"
  - "3.3"
  - "3.4"
+ - "3.5"
 
+# Twisted dropped Python 2.6 support, so install a compatible version on 2.6
 install:
+ - if [[ $TRAVIS_PYTHON_VERSION == 2.6 ]]; then pip install Twisted==12.2; fi
  - pip install -e .[commands,consumers]
- - pip install nose mock sqlalchemy unittest2
```
Comment (pypingou) [inline .travis.yml]: isn't unittest2 required in 2.6?

Comment (jeremycline) [inline .travis.yml]: It is, and it's placed in the tests_require list in setup.py

**Comment by jeremycline** [comment]
I added a ``test-requirements.txt`` file with all the dependencies so we can pip install them. Simply doing ``python setup.py nosetests`` to install the test requirements was causing installations to fail on 2.6 (for reasons I did not dig into too deeply).

**Comment by pypingou** [comment]
:+1: for me

**Pull request merged**